src/systemctl/systemctl-start-unit.c

```diff
@@ -399,3 +399,42 @@ int verb_start(int argc, char *argv[], void *userdata) {
 
         return ret;
 }
+
+int verb_toggle(int argc, char *argv[], void *userdata) {
+        _cleanup_strv_free_ char **names = NULL;
+        sd_bus *bus;
+        int r, ret = EXIT_SUCCESS;
+
+        if (!argv[1])
+                return 0;
+
+        r = mangle_names("to toggle", strv_skip(argv, 1), &names);
+        if (r < 0)
+                return r;
+
+        r = acquire_bus(BUS_MANAGER, &bus);
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(name, names) {
+                UnitActiveState active_state;
+                bool is_active = false;
+                char *new_args[3];
+
+                r = get_state_one_unit(bus, *name, &active_state);
+                if (r < 0)
+                        return r;
+
+                is_active = IN_SET(active_state, UNIT_ACTIVE, UNIT_RELOADING);
+
+                new_args[0] = (char*) (is_active ? "stop" : "start");
+                new_args[1] = basename(*name);
+                new_args[2] = NULL;
+
+                r = verb_start(2, new_args, userdata);
```

Comment:
We shouldn't limit ourselves to the interface of `verb_start` for this. It's interface is specifically designed to take arguments from the command line which is not what we're doing here.

What I'd do instead is to split out a common function out of `verb_start` that does the stuff that's needed for both `verb_start` and `verb_toggle` and to then call that function from both `verb_start` and `verb_toggle`.

The interface of this function would probably look something like:

```
int do_action(enum action action, const char **units)
```